Account.java

```diff
@@ -0,0 +1,93 @@
+package doubleEntry;
+
+import doubleEntryComponents.Agent;
+import doubleEntryComponents.actions.Action;
+import doubleEntryComponents.contracts.Contract;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+public class Account {
+
+    public Account(String name, AccountType accountType, Double startingBalance) {
+        this.name = name;
+        this.accountType = accountType;
+        this.total = startingBalance;
+        this.contractClass = null;
+        this.contracts = new HashSet<>();
+    }
+
+    public Account(String name, AccountType accountType) {
+        this(name,accountType,0.0);
+    }
+
+    private double total;
+
+    private Collateral collateralType;
+    private AccountType accountType;
+    private String name;
+    private Class<? extends Contract> contractClass;
+    protected HashSet<Contract> contracts;
+
+
+    public void addContract(Contract contract) {
+        contracts.add(contract);
+    }
+
+    /**
+     * A Debit is a positive change for ASSET and EXPENSES accounts, and negative for the rest.
+     * @param amount the amount to debit
+     */
+    public void debit(double amount) {
+        if ((accountType==AccountType.ASSET) || (accountType==AccountType.EXPENSES)) {
+            total += amount;
+        } else {
+            total -= amount;
+        }
+    }
+
+    /**
+     * A Credit is a negative change for ASSET and EXPENSES accounts, and positive for the rest.
+     * @param amount the amount to credit
+     */
+    public void credit(double amount) {
+        if ((accountType==AccountType.ASSET) || (accountType==AccountType.EXPENSES)) {
+            total -= amount;
+        } else {
+            total += amount;
+        }
+    }
+
+
+    public ArrayList<Action> getAvailableActions(Agent me) {
```

Comment:
I really like your idea that behaviour is defined as choosing an action(s) from a set of potential actions. I'm a little confused how it is being implemented. Perhaps we can discuss how you envision the interface for Contracts (and other classes) providing available actions. Is it only Contracts which provide potential actions? Should there be a general interface/class structure for actions and the selection of actions?